vault/resource_managed_keys.go

```diff
@@ -0,0 +1,473 @@
+		resp, err := client.Logical().Read(awsKeyPath)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		data := map[string]interface{}{}
+		for k := range managedKeysAWSConfigSchema() {
+			if v, ok := resp.Data[k]; ok {
+				data[k] = v
+			}
+
+			// set these from TF config since they won't
+			// be returned from Vault
+			if k == "access_key" || k == "secret_key" {
+				stateKey := fmt.Sprintf("%s.%d.%s", consts.FieldAWS, 0, k)
+				data[k] = d.Get(stateKey)
+			}
+		}
+		if err := d.Set(consts.FieldAWS, []map[string]interface{}{data}); err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	if _, ok := d.GetOk(consts.FieldPKCS); ok {
+		pkcsKeyName := getKeyNameFromConfig(d, consts.FieldPKCS)
+		pkcsKeyPath := getManagedKeysPath(KMSTypePKCS, pkcsKeyName)
+		resp, err := client.Logical().Read(pkcsKeyPath)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		data := map[string]interface{}{}
+		for k := range managedKeysPKCSConfigSchema() {
+			if v, ok := resp.Data[k]; ok {
+				data[k] = v
+			}
+
+			// set these from TF config since they won't
+			// be returned from Vault
+			if k == "pin" {
+				stateKey := fmt.Sprintf("%s.%d.%s", consts.FieldPKCS, 0, k)
+				data[k] = d.Get(stateKey)
+			}
+		}
+		if err := d.Set(consts.FieldPKCS, []map[string]interface{}{data}); err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	if _, ok := d.GetOk(consts.FieldAzure); ok {
+		azureKeyName := getKeyNameFromConfig(d, consts.FieldAzure)
+		azureKeyPath := getManagedKeysPath(KMSTypeAzure, azureKeyName)
+		resp, err := client.Logical().Read(azureKeyPath)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		data := map[string]interface{}{}
+		for k := range managedKeysAzureConfigSchema() {
+			if v, ok := resp.Data[k]; ok {
+				data[k] = v
+			}
+		}
+		if err := d.Set(consts.FieldAzure, []map[string]interface{}{data}); err != nil {
+			return diag.FromErr(err)
+		}
+
+	}
+
+	return nil
+}
+
+func managedKeysDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	if _, ok := d.GetOk(consts.FieldAWS); ok {
+		awsKeyName := getKeyNameFromConfig(d, consts.FieldAWS)
+		path := getManagedKeysPath(KMSTypeAWS, awsKeyName)
+
+		log.Printf("[DEBUG] Deleting managed key %s", path)
+		_, err := client.Logical().Delete(path)
+		if err != nil {
+			return diag.Errorf("error deleting managed key %s", path)
+		}
+		log.Printf("[DEBUG] Deleted managed key %q", path)
+	}
+
+	if _, ok := d.GetOk(consts.FieldAzure); ok {
+		azureKeyName := getKeyNameFromConfig(d, consts.FieldAzure)
+		path := getManagedKeysPath(KMSTypeAzure, azureKeyName)
+
+		log.Printf("[DEBUG] Deleting managed key %s", path)
+		_, err := client.Logical().Delete(path)
+		if err != nil {
+			return diag.Errorf("error deleting managed key %s", path)
+		}
+		log.Printf("[DEBUG] Deleted managed key %q", path)
+	}
+
```

Comment:
Do we support deleting the `PKCS` type?

Reply:
Yup! Happened to miss deletion for PKCS in previous commit, thanks for the catch! Added in latest commits